### PR TITLE
PE-58:requisition definition validation

### PR DESCRIPTION
**Pull request by ssalunkhe-opennms**
### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  If you're not sure how to do this, we can help you get things set up properly.
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/PE-58

### CODE REVIEW

**Review by scthompsonopen — CHANGES_REQUESTED** [review]
I've left a few small comments. Nothing major, but should help in the long term maintainability of the project.

File: ui/src/components/Configuration/ReqDefinitionForm.vue
Comment (ssalunkhe-opennms) [reply ui/src/components/Configuration/ReqDefinitionForm.vue:51]: Class Name Changed

File: ui/src/components/Configuration/ReqDefinitionForm.vue
Comment (ssalunkhe-opennms) [reply ui/src/components/Configuration/ReqDefinitionForm.vue:67]: Class Name Changed

File: ui/src/components/Configuration/ReqDefinitionForm.vue
Comment (ssalunkhe-opennms) [reply ui/src/components/Configuration/ReqDefinitionForm.vue:165]: Done

File: ui/src/components/Configuration/ReqDefinitionForm.vue
Comment (ssalunkhe-opennms) [reply ui/src/components/Configuration/ReqDefinitionForm.vue:169]: Done

File: ui/src/components/ValidationMessage.vue
Comment (ssalunkhe-opennms) [reply ui/src/components/ValidationMessage.vue:3]: Yes. We can use <li> tag with any other DOM element (<div>) without <ul> OR <ol>. 
Now, I changed this to <p> tag as suggested. 
Note : <span> tag not used because custom CSS required to add on it whether <p> provide built-in CSS for element.

File: ui/src/components/Configuration/ReqDefinitionForm.vue
Comment (ssalunkhe-opennms) [reply ui/src/components/Configuration/ReqDefinitionForm.vue:181]: Done

**Comment by mikewrosey** [comment]
Separating the custom validators makes sense but validators aren't really components, so I'm not sure about having the `Validators` folder inside `Components`. What do you guys think about moving that up a level?

Also, what is the intention of this state and models folders? Is this pattern documented somewhere? To me the model should be moved inside the component. I can't think of another scenario where we would import that elsewhere. Likewise for state. Was that intended to contain other states? If so, we'd be importing states we don't need into other components. Seems like this should be part of the ReqDefinitionForm as well. Thoughts? I know this makes the component larger but one way to reduce size is to use the `script setup` syntax like with the others. All components should be using that.

**Comment by ssalunkhe-opennms** [comment]
> Separating the custom validators makes sense but validators aren't really components, so I'm not sure about having the `Validators` folder inside `Components`. What do you guys think about moving that up a level?
> 
> Also, what is the intention of these state and models folders? Is this pattern documented somewhere? To me the model should be moved inside the component. I can't think of another scenario where we would import that elsewhere. Likewise for state. Was that intended to contain other states? If so, we'd be importing states we don't need into other components. Seems like this should be part of the ReqDefinitionForm as well. Thoughts? I know this makes the component larger but one way to reduce size is to use the `script setup` syntax like with the others. All components should be using that.

Below changes done -
- Setup added to script tag.
- Validators folder moved to src folder.
- Form Model moved inside the Configuration folder as it belongs to it only.
- formState(old name: State) folder moved to configuration folder to manage form state globally.

**Review by scthompsonopen — APPROVED** [review]
I've approved the changes you made for me. I'll let Mike respond in regards to his feedback, so please wait for his review before merging. :)

**Comment by mikewrosey** [comment]
Can I make a suggestion about architecture here..


> > Separating the custom validators makes sense but validators aren't really components, so I'm not sure about having the `Validators` folder inside `Components`. What do you guys think about moving that up a level?
> > Also, what is the intention of these state and models folders? Is this pattern documented somewhere? To me the model should be moved inside the component. I can't think of another scenario where we would import that elsewhere. Likewise for state. Was that intended to contain other states? If so, we'd be importing states we don't need into other components. Seems like this should be part of the ReqDefinitionForm as well. Thoughts? I know this makes the component larger but one way to reduce size is to use the `script setup` syntax like with the others. All components should be using that.
> 
> Below changes done -
> 
> * Setup added to script tag.
> * Validators folder moved to src folder.
> * Form Model moved inside the Configuration folder as it belongs to it only.
> * formState(old name: State) folder moved to configuration folder to manage form state globally.

Alright, well if it's contained inside the configuration folder, I think that's better. 
One last thing is that the ValidationMessage.vue component should be moved either to Common or Configuration, depending on the intention, but I'll let you do that at your leisure and approve this one.

**Review by mikewrosey — APPROVED** [review]

**Pull request merged**
